#!/usr/bin/env python
PACKAGE = "simulation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("laser_obstacle_color", int_t,   0, "color of a pixel on an occupancy grid that should be regarded as an obstacle", 0, 0, 255)
gen.add("laser_max_sensor_distance", double_t,   0, "maximum sensing distance of the simulated laser range sensor", 10.0, 0, 30.0)
gen.add("laser_min_sensor_distance", double_t,   0, "minimum sensing distance of the simulated laser range sensor", 0, 0, 30.0)
gen.add("laser_field_of_view", double_t,   0, "the simulated laser range sensors field of view (deg)", 70.0, 0, 360)
gen.add("laser_angular_resolution", double_t,   0, "angular resolution of the laser sensors field of view (deg)", 0.2, 0.01, 360)

gen.add("us_obstacle_color", int_t,   0, "color of a pixel on an occupancy grid that should be regarded as an obstacle", 0, 0, 255)
gen.add("us_max_sensor_distance", double_t,   0, "maximum sensing distance of the simulated us range sensor", 2.7, 0, 30.0)
gen.add("us_min_sensor_distance", double_t,   0, "minimum sensing distance of the simulated us range sensor", 0, 0, 30.0)
gen.add("us_field_of_view", double_t,   0, "the simulated us range sensors field of view (deg)", 45.0, 0, 360)
gen.add("us_angular_resolution", double_t,   0, "angular resolution of the us sensors field of view (deg)", 0.5, 0.01, 360)

exit(gen.generate(PACKAGE, "simulation", "RangeSensor"))
